### PR TITLE
Add table aws_wafv2_rule_group. closes #278

### LINKED ISSUE #278: Add table aws_wafv2_rule_group
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go/service/wafv2

### PR DESCRIPTION
**Pull request by sayan133**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_wafv2_rule_group []

PRETEST: tests/aws_wafv2_rule_group

TEST: tests/aws_wafv2_rule_group
Running terraform
aws_wafv2_rule_group.named_test_resource_regional: Creating...
aws_wafv2_rule_group.named_test_resource_global: Creating...
aws_wafv2_rule_group.named_test_resource_global: Creation complete after 4s [id=a56ac148-008b-4510-85b6-09d830da4c82]
aws_wafv2_rule_group.named_test_resource_regional: Creation complete after 4s [id=bc6471ac-6d2d-4c18-b4ed-d51b30935986]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

account_id = "388460667113"
aws_partition = "aws"
aws_region = "us-east-1"
resource_aka_global = "arn:aws:wafv2:us-east-1:388460667113:global/rulegroup/turbottest8542_global/a56ac148-008b-4510-85b6-09d830da4c82"
resource_aka_regional = "arn:aws:wafv2:us-east-1:388460667113:regional/rulegroup/turbottest8542_regional/bc6471ac-6d2d-4c18-b4ed-d51b30935986"
resource_id_global = "a56ac148-008b-4510-85b6-09d830da4c82"
resource_id_regional = "bc6471ac-6d2d-4c18-b4ed-d51b30935986"
resource_name_global = "turbottest8542_global"
resource_name_regional = "turbottest8542_regional"

Running SQL query: test-getGlobal-query.sql
[
  {
    "account_id": "388460667113",
    "arn": "arn:aws:wafv2:us-east-1:388460667113:global/rulegroup/turbottest8542_global/a56ac148-008b-4510-85b6-09d830da4c82",
    "capacity": 2,
    "description": "A global rule group for testing.",
    "id": "a56ac148-008b-4510-85b6-09d830da4c82",
    "name": "turbottest8542_global",
    "partition": "aws",
    "region": "global",
    "rules": [
      {
        "Action": {
          "Allow": {},
          "Block": null,
          "Count": null
        },
        "Name": "rule-1",
        "OverrideAction": null,
        "Priority": 1,
        "Statement": {
          "AndStatement": null,
          "ByteMatchStatement": null,
          "GeoMatchStatement": {
            "CountryCodes": [
              "US",
              "IN"
            ],
            "ForwardedIPConfig": null
          },
          "IPSetReferenceStatement": null,
          "ManagedRuleGroupStatement": null,
          "NotStatement": null,
          "OrStatement": null,
          "RateBasedStatement": null,
          "RegexPatternSetReferenceStatement": null,
          "RuleGroupReferenceStatement": null,
          "SizeConstraintStatement": null,
          "SqliMatchStatement": null,
          "XssMatchStatement": null
        },
        "VisibilityConfig": {
          "CloudWatchMetricsEnabled": false,
          "MetricName": "turbottest8542_global",
          "SampledRequestsEnabled": false
        }
      }
    ],
    "scope": "CLOUDFRONT",
    "visibility_config": {
      "CloudWatchMetricsEnabled": false,
      "MetricName": "turbottest8542_global",
      "SampledRequestsEnabled": false
    }
  }
]
✔ PASSED

Running SQL query: test-getRegional-query.sql
[
  {
    "account_id": "388460667113",
    "arn": "arn:aws:wafv2:us-east-1:388460667113:regional/rulegroup/turbottest8542_regional/bc6471ac-6d2d-4c18-b4ed-d51b30935986",
    "capacity": 2,
    "description": "A regional rule group for testing.",
    "id": "bc6471ac-6d2d-4c18-b4ed-d51b30935986",
    "name": "turbottest8542_regional",
    "partition": "aws",
    "region": "us-east-1",
    "rules": [
      {
        "Action": {
          "Allow": {},
          "Block": null,
          "Count": null
        },
        "Name": "rule-1",
        "OverrideAction": null,
        "Priority": 1,
        "Statement": {
          "AndStatement": null,
          "ByteMatchStatement": null,
          "GeoMatchStatement": {
            "CountryCodes": [
              "US",
              "IN"
            ],
            "ForwardedIPConfig": null
          },
          "IPSetReferenceStatement": null,
          "ManagedRuleGroupStatement": null,
          "NotStatement": null,
          "OrStatement": null,
          "RateBasedStatement": null,
          "RegexPatternSetReferenceStatement": null,
          "RuleGroupReferenceStatement": null,
          "SizeConstraintStatement": null,
          "SqliMatchStatement": null,
          "XssMatchStatement": null
        },
        "VisibilityConfig": {
          "CloudWatchMetricsEnabled": false,
          "MetricName": "turbottest8542_regional",
          "SampledRequestsEnabled": false
        }
      }
    ],
    "scope": "REGIONAL",
    "visibility_config": {
      "CloudWatchMetricsEnabled": false,
      "MetricName": "turbottest8542_regional",
      "SampledRequestsEnabled": false
    }
  }
]
✔ PASSED

Running SQL query: test-hydrateGlobal-query.sql
[
  {
    "description": "A global rule group for testing.",
    "name": "turbottest8542_global",
    "tags_src": [
      {
        "Key": "scope",
        "Value": "global"
      },
      {
        "Key": "foo1",
        "Value": "bar1"
      }
    ]
  }
]
✔ PASSED

Running SQL query: test-hydrateRegional-query.sql
[
  {
    "description": "A regional rule group for testing.",
    "name": "turbottest8542_regional",
    "tags_src": [
      {
        "Key": "foo",
        "Value": "bar"
      },
      {
        "Key": "scope",
        "Value": "regional"
      }
    ]
  }
]
✔ PASSED

Running SQL query: test-listGlobal-query.sql
[
  {
    "arn": "arn:aws:wafv2:us-east-1:388460667113:global/rulegroup/turbottest8542_global/a56ac148-008b-4510-85b6-09d830da4c82",
    "id": "a56ac148-008b-4510-85b6-09d830da4c82",
    "name": "turbottest8542_global",
    "region": "global",
    "scope": "CLOUDFRONT"
  }
]
✔ PASSED

Running SQL query: test-listRegional-query.sql
[
  {
    "arn": "arn:aws:wafv2:us-east-1:388460667113:regional/rulegroup/turbottest8542_regional/bc6471ac-6d2d-4c18-b4ed-d51b30935986",
    "id": "bc6471ac-6d2d-4c18-b4ed-d51b30935986",
    "name": "turbottest8542_regional",
    "region": "us-east-1",
    "scope": "REGIONAL"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:wafv2:us-east-1:388460667113:regional/rulegroup/turbottest8542_regional/bc6471ac-6d2d-4c18-b4ed-d51b30935986"
    ],
    "tags": {
      "foo": "bar",
      "scope": "regional"
    },
    "title": "turbottest8542_regional"
  }
]
✔ PASSED

POSTTEST: tests/aws_wafv2_rule_group

TEARDOWN: tests/aws_wafv2_rule_group

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
 
### Basic info
```sql
select
  name,
  description,
  arn,
  id,
  scope,
  capacity,
  rules,
  region
from
  aws_wafv2_rule_group;
```
```
+-------------------------+-------------------------------------+----------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+----------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| name                    | description                         | arn                                                                                                                  | id                                   | scope      | capacity | rules                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | region    |
+-------------------------+-------------------------------------+----------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+----------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
| test_steampipe_regional | This is a test regional rule group. | arn:aws:wafv2:us-east-1:388460667113:regional/rulegroup/test_steampipe_regional/d437f495-207e-481a-8b08-5fa4428c8c8b | d437f495-207e-481a-8b08-5fa4428c8c8b | REGIONAL   | 5        | [{"Action":{"Allow":null,"Block":{},"Count":null},"Name":"rule-1","OverrideAction":null,"Priority":0,"Statement":{"AndStatement":null,"ByteMatchStatement":{"FieldToMatch":{"AllQueryArguments":null,"Body":null,"JsonBody":null,"Method":{},"QueryString":null,"SingleHeader":null,"SingleQueryArgument":null,"UriPath":null},"PositionalConstraint":"EXACTLY","SearchString":"UFVU","TextTransformations":[{"Priority":0,"Type":"NONE"}]},"GeoMatchStatement":null,"IPSetReferenceStatement":null,"ManagedRuleGroupStatement":null,"NotStatement":null,"OrStatement":null,"RateBasedStatement":null,"RegexPatternSetReferenceStatement":null,"RuleGroupReferenceStatement":null,"SizeConstraintStatement":null,"SqliMatchStatement":null,"XssMatchStatement":null},"VisibilityConfig":{"CloudWatchMetricsEnabled":true,"MetricName":"rule-1","SampledRequestsEnabled":true}}] | us-east-1 |
| test_steampipe_global   | This is a global rule group.        | arn:aws:wafv2:us-east-1:388460667113:global/rulegroup/test_steampipe_global/daf509d5-5b2a-4c72-9f7d-d309e51baccd     | daf509d5-5b2a-4c72-9f7d-d309e51baccd | CLOUDFRONT | 2        | [{"Action":{"Allow":null,"Block":{},"Count":null},"Name":"rule-1","OverrideAction":null,"Priority":0,"Statement":{"AndStatement":null,"ByteMatchStatement":{"FieldToMatch":{"AllQueryArguments":null,"Body":null,"JsonBody":null,"Method":{},"QueryString":null,"SingleHeader":null,"SingleQueryArgument":null,"UriPath":null},"PositionalConstraint":"EXACTLY","SearchString":"UFVU","TextTransformations":[{"Priority":0,"Type":"NONE"}]},"GeoMatchStatement":null,"IPSetReferenceStatement":null,"ManagedRuleGroupStatement":null,"NotStatement":null,"OrStatement":null,"RateBasedStatement":null,"RegexPatternSetReferenceStatement":null,"RuleGroupReferenceStatement":null,"SizeConstraintStatement":null,"SqliMatchStatement":null,"XssMatchStatement":null},"VisibilityConfig":{"CloudWatchMetricsEnabled":true,"MetricName":"rule-1","SampledRequestsEnabled":true}}] | global    |
+-------------------------+-------------------------------------+----------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+----------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
```

### List global (CLOUDFRONT) Rule Groups
```sql
select
  name,
  description,
  arn,
  id,
  scope,
  capacity,
  region
from
  aws_wafv2_rule_group
where
  scope = 'CLOUDFRONT';
```

```
+-----------------------+------------------------------+------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+----------+--------+
| name                  | description                  | arn                                                                                                              | id                                   | scope      | capacity | region |
+-----------------------+------------------------------+------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+----------+--------+
| test_steampipe_global | This is a global rule group. | arn:aws:wafv2:us-east-1:388460667113:global/rulegroup/test_steampipe_global/daf509d5-5b2a-4c72-9f7d-d309e51baccd | daf509d5-5b2a-4c72-9f7d-d309e51baccd | CLOUDFRONT | 2        | global |
+-----------------------+------------------------------+------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+----------+--------+
```

### List Rule Groups having having capacity less than 5

```sql
select
  name,
  description,
  arn,
  id,
  scope,
  capacity,
  region
from
  aws_wafv2_rule_group
where
  capacity < 5;
```

```
+-----------------------+------------------------------+------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+----------+--------+
| name                  | description                  | arn                                                                                                              | id                                   | scope      | capacity | region |
+-----------------------+------------------------------+------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+----------+--------+
| test_steampipe_global | This is a global rule group. | arn:aws:wafv2:us-east-1:388460667113:global/rulegroup/test_steampipe_global/daf509d5-5b2a-4c72-9f7d-d309e51baccd | daf509d5-5b2a-4c72-9f7d-d309e51baccd | CLOUDFRONT | 2        | global |
+-----------------------+------------------------------+------------------------------------------------------------------------------------------------------------------+--------------------------------------+------------+----------+--------+
```
</details>

### CODE REVIEW

File: aws/table_aws_wafv2_rule_group.go [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+				Transform:   transform.From(regexPatternSetLocation),
+			},
+			{
+				Name:        "description",
+				Description: "A description of the rule group that helps with identification.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Description", "RuleGroup.Description"),
+			},
+			{
+				Name:        "capacity",
```
Comment (khushboo9024) [inline aws/table_aws_wafv2_rule_group.go:61]: `capacity` column should be above `description` column.

File: aws/table_aws_wafv2_rule_group.go [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+	ruleGroupTags, err := svc.ListTagsForResource(param)
+	if err != nil {
+		return nil, err
+	}
+	return ruleGroupTags, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func regexPatternSetLocation(_ context.Context, d *transform.TransformData) (interface{}, error) {
```
Comment (khushboo9024) [inline aws/table_aws_wafv2_rule_group.go:296]:
```suggestion
func ruleGroupLocation(_ context.Context, d *transform.TransformData) (interface{}, error) {
```

File: aws/table_aws_wafv2_rule_group.go [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+			},
+			{
+				Name:        "id",
+				Description: "A unique identifier for the rule group.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Id", "RuleGroup.Id"),
+			},
+			{
+				Name:        "scope",
+				Description: "Specifies the scope of the Regex Pattern Set. Possible values are: 'REGIONAL' and 'CLOUDFRONT'.",
```
Comment (khushboo9024) [inline aws/table_aws_wafv2_rule_group.go:50]: Description is wrong.

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Minor changes required.

**Review by khushboo9024 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — CHANGES_REQUESTED** [review]
Made quick remark to change.

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**